immutable.d.ts

```diff
@@ -1960,8 +1974,8 @@ export function takeUntil<T>(predicate: (x: T) => boolean): (list: readonly T[])
  * Note, that you cannot use `update` operation, if the object path is missing in the input object.
  * Also, you cannot use `add` operation, if the object path has a value.
  */
-export function applyDiff<Output>(rules: readonly ApplyDiffRule[], obj: object): Output;
-export function applyDiff<Output>(rules: readonly ApplyDiffRule[]): ( obj: object) => Output;
+export function applyDiff<Output, Input extends object>(rules: readonly ApplyDiffRule<Paths<Input>>[], obj: Input): Output;
```

Comment:
I am surprised that TS doesn't complain about this. I think `Record<string, unknown>` is the recommended replacement of `object`.